### PR TITLE
1307 - activity logging

### PR DESCRIPTION
**Pull request by HalcyonJAC**
**Author checklist**

- [ ] Include primary ticket number in title - e.g. "#123 New styling for widget" - and any additional tickets in description
- [ ] Fill in the details below and delete as appropriate
- [ ] Be proactive in getting your work approved 💪

---

## What's included?
Describe the changes included in this pull request and highlight dependencies with other repos/tickets

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?
Describe the steps required to test & verify this change

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work
🟠 Medium risk - this does change code that is shared with other areas
🔴 High risk - this includes a lot of changes to shared code

## Additional context
Include screen grabs, notes etc.

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

### CODE REVIEW

**Comment by warrensearle** [comment]
@HalcyonJAC @JeremykJAC is this one still relevant?

**Comment by HalcyonJAC** [comment]
> @HalcyonJAC @JeremykJAC is this one still relevant?

@warrensearle I have merged the main branch into this one, resolved all conflicts and done some tests of my own.

**Review by warrensearle — APPROVED** [review]
Code changes look fine 👍

File: src/views/Apply/ApplyMixIn.js [hunk 1]
```diff
@@ -128,8 +131,12 @@ export default {
       if (this.isValid() && this.formId) {
         this.formData.progress[this.formId] = true;
         await this.$store.dispatch('application/save', this.formData);
+        this.logEventAfterSave();
         this.$router.push({ name: 'task-list' });
       }
     },
+    logEventAfterSave() {
+      // do nothing - can be overriden in parent class
+    },
```
Comment (tomlovesgithub) [inline src/views/Apply/ApplyMixIn.js:140]: Why can't this whole function be removed?

**Review by tomlovesgithub — APPROVED** [review]

File: src/views/Apply/ApplyMixIn.js (hunk 1 above)
Comment (HalcyonJAC) [reply src/views/Apply/ApplyMixIn.js:140]: The "save" method will call the "logEventAfterSave" method which will be implemented in other classes. So keeping it here can make the code more readable.

**Pull request merged**